### DISCUSSION TITLE
splitting json payload based on the field names

### DISCUSSION THREAD

**Discussion by sesa657547** [redpanda-data/connect#2262] [Q&A]
Hi,

I am new to benthos and on the process of learning it. I have got a situation where I need to split the single payload into multiple messages based on the field names and send these messaged to different kafka topics. My payload looks like this:

[
{
"dId": "kjglsjglfkgfdkgfhjfhdsj",
"d1": "fhkfjsaf",
"d2": {
"dc1": 8,
"dc2": 9
},
"timestamp": "2020-09-03",
"A": [
{
"aId": "jflksflskgskg",
"af1": "kkfls",
"af2": {
"ac1": 6,
"ac2": 2
},
"af3": 1,

      "af4": "00",
      "af5": 4
    }
  ],
  "B": [
    {
      "bId": "jgljgldjgs",
      "bf1": 6,
      "bf2": {
        "bc1": 4,
        "bc2": 2
      }
    }
  ],
  "C": [
    {
      "cId": "njfglsjg",
      "cf1": 3,
      "cf2": "jfgg"
    }
  ]
}
]

I want to send A array data to one kafka topics and similarly for B and C.
Thanks
M

**Comment by mihaitodor** [comment]
This is a duplicate of #2252. Please see my answer in that thread and follow up there if you have more questions.
